paasta_tools/cli/cmds/logs.py

```diff
@@ -559,6 +624,65 @@ def cluster_to_scribe_env(self, cluster):
             return env
 
 
+def generate_start_end_time(time_ago="30m", time=datetime.datetime.utcnow()):
+    time_ago = timeparse(time_ago)
+    start_time = time - datetime.timedelta(seconds=time_ago)
+    end_time = time
+
+    # Covert the timestamps to something timezone aware
+    start_time = pytz.utc.localize(start_time)
+    end_time = pytz.utc.localize(end_time)
+
+    return start_time, end_time
+
+
+def validate_filtering_args(args, log_reader):
+    if not log_reader.SUPPORTS_LINE_OFFSET and args.line_offset is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line based offsets")
+        return False
+    if not log_reader.SUPPORTS_LINE_COUNT and args.line_count is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line count based log retrieval")
+        return False
+    if not log_reader.SUPPORTS_TAILING and args.tail:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support tailing")
+        return False
+    if not log_reader.SUPPORTS_TIME and (args.time is not None or args.time_ago is not None):
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support time based offsets")
+        return False
+
+    if args.tail and (args.line_count is not None or args.time_ago is not None or
+                      args.time is not None or args.line_offset is not None):
+        print PaastaColors.red("You cannot specify line/time based filtering parameters when tailing")
+        return False
+
+    # Can't have both
+    if args.line_count is not None and args.time_ago is not None:
+        print PaastaColors.red("You cannot filter based on both line counts and time")
+        return False
+
+    return True
+
+
+def pick_default_log_mode(args, log_reader, service, levels, components, clusters):
+    if log_reader.SUPPORTS_LINE_COUNT:
+        sys.stderr.write(PaastaColors.cyan("No filtering specified, grabbing last 100 lines") + "\n")
+        log_reader.print_last_n_logs(service, 100, levels, components, clusters, raw_mode=args.raw_mode)
+        return 0
+
+    elif log_reader.SUPPORTS_TIME:
+        start_time, end_time = generate_start_end_time()
+
+        sys.stderr.write(PaastaColors.cyan("No filtering specified, grabbing last 30 minutes of logs") + "\n")
+        log_reader.print_logs_by_time(service, start_time, end_time, levels, components, clusters,
+                                      raw_mode=args.raw_mode)
+        return 0
+
+    elif log_reader.SUPPORTS_TAILING:
+        sys.stderr.write(PaastaColors.cyan("No filtering specified, tailing logs") + "\n")
+        log_reader.tail_logs(service, levels, components, clusters, raw_mode=args.raw_mode)
+        return 0
```

Comment:
The reason for printing them to stderr is in case someone wants to pipe their logs out to a file for example. If we printed it to stdout, this output would get mangled in along with it.